src/main/java/ladder/model/Users.java

```diff
@@ -0,0 +1,31 @@
+package ladder.model;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Users {
+
+    private final List<User> users;
+
+    public Users(List<User> users) {
+        this.users = users;
+    }
+
+    public static Users of(List<String> userNames) {
+        List<User> userList = userNames.stream()
+                                    .map(User::new)
+                                    .collect(Collectors.toList());
+        return new Users(userList);
```

Comment:
```suggestion
       return userNames.stream()
                .map(User::new)
                .collect(Collectors.collectingAndThen(Collectors.toList(), Users::new));
```
`collectingAndThen`을 활용해볼수 있을것 같아요 !